torchgen/utils.py

```diff
@@ -396,3 +396,52 @@ def _format(
     indent_str = " " * indent
     body = f", {delimiter}{curr_indent_str}".join(fields_str)
     return f"{start}{indent_str}{body}{end}"
+
+
+class NamespaceHelper:
```

Comment:
This class might be a good place to check that the namespace provided in the YAML file is a valid C++ namespace; i.e., elements are in `/[a-zA-Z_][a-zA-Z_0-9]*/`. That would help trigger error early, when operating on the YAML files, instead of later on when building the generated code files.